kafka-connect-s3/src/main/java/io/confluent/connect/s3/storage/S3Storage.java

```diff
@@ -101,16 +106,45 @@ public ClientConfiguration newClientConfiguration(S3SinkConnectorConfig config)
     clientConfiguration.withUserAgentPrefix(version);
     if (StringUtils.isNotBlank(config.getString(S3_PROXY_URL_CONFIG))) {
       S3ProxyConfig proxyConfig = new S3ProxyConfig(config);
+      RetryPolicy fullJitterPolicy = newRetryPolicy(config);
       clientConfiguration.withProtocol(proxyConfig.protocol())
           .withProxyHost(proxyConfig.host())
           .withProxyPort(proxyConfig.port())
           .withProxyUsername(proxyConfig.user())
-          .withProxyPassword(proxyConfig.pass());
+          .withProxyPassword(proxyConfig.pass())
+          .withRetryPolicy(fullJitterPolicy);
     }
 
     return clientConfiguration;
   }
 
+
+  /**
+   * Creates a retry policy, based on full jitter backoff strategy
+   * and default retry condition.
```

Comment:
Add "This method currently configures the AWS client retry policy to use full jitter."

Reply:
Do you mean to add this to the method `io.confluent.connect.s3.storage.S3Storage#newS3Client`?

Reply:
Renaming this method helps, and means my JavaDoc suggestion isn't really needed since the method name explains that it's using full jitter. So you can disregard this comment.

Reply:
I've added Javadocs to those methods just in case.